gd32l233r_eval fails to build sensor_shell sample with async

During the introduction of https://github.com/zephyrproject-rtos/zephyr/pull/60063 the CI was failing to build `gd32l233r_eval` with an assembler error which I believe is a part of the toolchain (see below for the build log). To replicate, simply remove the `gd32l233r_eval` from the excluded platforms list.

```
INFO    - 166/671 gd32l233r_eval            samples/sensor/sensor_shell/sample.sensor.shell     ERROR Build failure (build)
INFO    - /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/build.log
ERROR   - Loading Zephyr module(s) (Zephyr base): zephyr_default:kconfig
-- Application: /__w/zephyr/zephyr/samples/sensor/sensor_shell
-- CMake version: 3.22.1
-- Found Python3: /usr/bin/python (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /github/home/.cache/zephyr
-- Zephyr version: 3.5.0-rc1 (/__w/zephyr/zephyr)
-- Found west (found suitable version "1.1.0", minimum required is "0.14.0")
-- Board: gd32l233r_eval
-- Found host-tools: zephyr 0.16.3 (/opt/toolchains/zephyr-sdk-0.16.3)
-- Found toolchain: zephyr 0.16.3 (/opt/toolchains/zephyr-sdk-0.16.3)
-- Found Dtc: /opt/toolchains/zephyr-sdk-0.16.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/gd32l233r_eval/gd32l233r_eval.dts
-- Generated zephyr.dts: /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/dts.cmake
Parsing /__w/zephyr/zephyr/samples/sensor/sensor_shell/Kconfig
Loaded configuration '/__w/zephyr/zephyr/boards/arm/gd32l233r_eval/gd32l233r_eval_defconfig'
Merged configuration '/__w/zephyr/zephyr/samples/sensor/sensor_shell/prj.conf'
Merged configuration '/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/misc/generated/extra_kconfig_options.conf'
Configuration saved to '/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/.config'
Kconfig header saved to '/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated/autoconf.h'
Loading Zephyr default modules (Zephyr base).
-- Application: /__w/zephyr/zephyr/samples/sensor/sensor_shell
-- CMake version: 3.22.1
-- Found Python3: /usr/bin/python (found suitable version "3.10.12", minimum required is "3.8") found components: Interpreter 
-- Cache files will be written to: /github/home/.cache/zephyr
-- Zephyr version: 3.5.0-rc1 (/__w/zephyr/zephyr)
-- Found west (found suitable version "1.1.0", minimum required is "0.14.0")
-- Board: gd32l233r_eval
-- Found host-tools: zephyr 0.16.3 (/opt/toolchains/zephyr-sdk-0.16.3)
-- Found toolchain: zephyr 0.16.3 (/opt/toolchains/zephyr-sdk-0.16.3)
-- Found Dtc: /opt/toolchains/zephyr-sdk-0.16.3/sysroots/x86_64-pokysdk-linux/usr/bin/dtc (found suitable version "1.6.0", minimum required is "1.4.6") 
-- Found BOARD.dts: /__w/zephyr/zephyr/boards/arm/gd32l233r_eval/gd32l233r_eval.dts
-- Generated zephyr.dts: /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/zephyr.dts
-- Generated devicetree_generated.h: /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated/devicetree_generated.h
-- Including generated dts.cmake file: /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/dts.cmake
Parsing /__w/zephyr/zephyr/samples/sensor/sensor_shell/Kconfig
Loaded configuration '/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/.config'
No change to configuration in '/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/.config'
No change to Kconfig header in '/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated/autoconf.h'
-- Found GnuLd: /opt/toolchains/zephyr-sdk-0.16.3/arm-zephyr-eabi/bin/../lib/gcc/arm-zephyr-eabi/12.2.0/../../../../arm-zephyr-eabi/bin/ld.bfd (found version "2.38") 
-- The C compiler identification is GNU 12.2.0
-- The CXX compiler identification is GNU 12.2.0
-- The ASM compiler identification is GNU
-- Found assembler: /opt/toolchains/zephyr-sdk-0.16.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc
-- Configuring done
-- Generating done
CMake Warning:
  Manually-specified variables were not used by the project:

    TC_RUNID


-- Build files have been written to: /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell
[1/156] Preparing syscall dependency handling

[2/156] Generating include/generated/version.h
-- Zephyr version: 3.5.0-rc1 (/__w/zephyr/zephyr), build: v3.5.0-rc1-33-gac6ddeb4a2ea
[3/156] Generating misc/generated/syscalls_subdirs.trigger
[4/156] Generating misc/generated/syscalls.json, misc/generated/struct_tags.json
[5/156] Generating include/generated/kobj-types-enum.h, include/generated/otype-to-str.h, include/generated/otype-to-size.h
[6/156] Generating include/generated/driver-validation.h
[7/156] Generating include/generated/syscall_dispatch.c, include/generated/syscall_list.h
[8/156] Building C object zephyr/CMakeFiles/offsets.dir/arch/arm/core/offsets/offsets.c.obj
[9/156] Generating include/generated/offsets.h
[10/156] Building C object CMakeFiles/app.dir/src/main.c.obj
[11/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/dec.c.obj
[12/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/hex.c.obj
[13/156] Linking C static library app/libapp.a
[14/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/multi_heap.c.obj
[15/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/posix/getopt/getopt_common.c.obj
[16/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/sem.c.obj
[17/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/timeutil.c.obj
[18/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/printk.c.obj
[19/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/posix/getopt/getopt.c.obj
[20/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/thread_entry.c.obj
[21/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/assert.c.obj
[22/156] Building C object zephyr/CMakeFiles/zephyr.dir/misc/generated/configs.c.obj
[23/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/bitarray.c.obj
[24/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mem_blocks.c.obj
[25/156] Building C object zephyr/CMakeFiles/zephyr.dir/soc/arm/gigadevice/gd32l23x/soc.c.obj
[26/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/tracing/tracing_none.c.obj
[27/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/ring_buffer.c.obj
[28/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/rb.c.obj
[29/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_cache.c.obj
[30/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_packaged.c.obj
[31/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap-validate.c.obj
[32/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/heap.c.obj
[33/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/cbprintf_complete.c.obj
[34/156] Building C object zephyr/CMakeFiles/zephyr.dir/lib/os/mpsc_pbuf.c.obj
[35/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_mgmt.c.obj
[36/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_output.c.obj
[37/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_msg.c.obj
[38/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_cmds.c.obj
[39/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/device_service.c.obj
[40/156] Building ASM object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/nmi_on_reset.S.obj
[41/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/kernel_service.c.obj
[42/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_fprintf.c.obj
[43/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/exc_exit.S.obj
[44/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_log_backend.c.obj
[45/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/backends/shell_uart.c.obj
[46/156] Building C object zephyr/arch/common/CMakeFiles/arch__common.dir/sw_isr_common.c.obj
[47/156] Generating linker_zephyr_pre0.cmd
[48/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_help.c.obj
[49/156] Building C object zephyr/arch/common/CMakeFiles/isr_tables.dir/isr_tables.c.obj
[50/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/logging/log_core.c.obj
[51/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fault_s.S.obj
[52/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/modules/devmem_service.c.obj
[53/156] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/tls.c.obj
[54/156] Linking C static library zephyr/arch/common/libisr_tables.a
[55/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_history.c.obj
[56/156] Building C object zephyr/CMakeFiles/zephyr_pre0.dir/misc/empty_file.c.obj
[57/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/reset.S.obj
[58/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_wildcard.c.obj
[59/156] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/nmi.c.obj
[60/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/vector_table.S.obj
[61/156] Linking C static library zephyr/arch/common/libarch__common.a
[62/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/irq_init.c.obj
[63/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fpu.c.obj
[64/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/scb.c.obj
[65/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_utils.c.obj
[66/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_cmds.c.obj
[67/156] Building C object zephyr/arch/arch/arm/core/CMakeFiles/arch__arm__core.dir/fatal.c.obj
[68/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/__aeabi_read_tp.S.obj
[69/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell_ops.c.obj
[70/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/swap_helper.S.obj
[71/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/cpu_idle.S.obj
[72/156] Building C object zephyr/CMakeFiles/zephyr.dir/subsys/shell/shell.c.obj
[73/156] Building ASM object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/isr_wrapper.S.obj
[74/156] Linking C static library zephyr/arch/arch/arm/core/libarch__arm__core.a
[75/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/thread_abort.c.obj
[76/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/fault.c.obj
[77/156] Linking C static library zephyr/libzephyr.a
[78/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/swap.c.obj
[79/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/irq_manage.c.obj
[80/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/prep_c.c.obj
[81/156] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/abort.c.obj
[82/156] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/common.c.obj
[83/156] Building C object zephyr/lib/posix/CMakeFiles/lib__posix.dir/fnmatch.c.obj
[84/156] Building C object zephyr/drivers/clock_control/CMakeFiles/drivers__clock_control.dir/clock_control_gd32.c.obj
[85/156] Building C object zephyr/lib/libc/picolibc/CMakeFiles/lib__libc__picolibc.dir/libc-hooks.c.obj
[86/156] Building C object zephyr/arch/arch/arm/core/cortex_m/cmse/CMakeFiles/arch__arm__core__cortex_m__cmse.dir/arm_core_cmse.c.obj
[87/156] Linking C static library zephyr/lib/posix/liblib__posix.a
[88/156] Building C object zephyr/arch/arch/arm/core/cortex_m/CMakeFiles/arch__arm__core__cortex_m.dir/thread.c.obj
[89/156] Linking C static library zephyr/arch/arch/arm/core/cortex_m/cmse/libarch__arm__core__cortex_m__cmse.a
[90/156] Linking C static library zephyr/drivers/clock_control/libdrivers__clock_control.a
[91/156] Building C object zephyr/drivers/interrupt_controller/CMakeFiles/drivers__interrupt_controller.dir/intc_gd32_exti.c.obj
[92/156] Linking C static library zephyr/lib/libc/picolibc/liblib__libc__picolibc.a
[93/156] Building C object zephyr/drivers/console/CMakeFiles/drivers__console.dir/uart_console.c.obj
[94/156] Building C object zephyr/drivers/pinctrl/CMakeFiles/drivers__pinctrl.dir/pinctrl_gd32_af.c.obj
[95/156] Linking C static library zephyr/arch/arch/arm/core/cortex_m/libarch__arm__core__cortex_m.a
[96/156] Linking C static library zephyr/drivers/interrupt_controller/libdrivers__interrupt_controller.a
[97/156] Building C object zephyr/lib/libc/common/CMakeFiles/lib__libc__common.dir/source/stdlib/malloc.c.obj
[98/156] Building C object zephyr/subsys/rtio/CMakeFiles/subsys__rtio.dir/rtio_init.c.obj
[99/156] Building C object zephyr/drivers/reset/CMakeFiles/drivers__reset.dir/reset_gd32.c.obj
[100/156] Linking C static library zephyr/drivers/pinctrl/libdrivers__pinctrl.a
[101/156] Linking C static library zephyr/drivers/console/libdrivers__console.a
[102/156] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/sys_clock_init.c.obj
[103/156] Building C object zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_decoders_init.c.obj
[104/156] Linking C static library zephyr/lib/libc/common/liblib__libc__common.a
[105/156] Building C object zephyr/subsys/rtio/CMakeFiles/subsys__rtio.dir/rtio_executor.c.obj
[106/156] Building C object modules/hal_gigadevice/CMakeFiles/hal_gigadevice.dir/__w/zephyr/modules/hal/gigadevice/gd32l23x/standard_peripheral/source/gd32l23x_dbg.c.obj
[107/156] Building C object zephyr/drivers/serial/CMakeFiles/drivers__serial.dir/usart_gd32.c.obj
[108/156] Linking C static library zephyr/drivers/reset/libdrivers__reset.a
[109/156] Building C object zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_shell_stream.c.obj
FAILED: zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_shell_stream.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.16.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DGD32L233 -DIRC32K_VALUE=32000 -DKERNEL -DPICOLIBC_DOUBLE_PRINTF_SCANF -D_FORTIFY_SOURCE=1 -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated -I/__w/zephyr/zephyr/soc/arm/gigadevice/gd32l23x -I/__w/zephyr/zephyr/lib/posix/getopt/. -I/__w/zephyr/zephyr/soc/arm/gigadevice/common/. -I/__w/zephyr/zephyr/soc/arm/gigadevice/gd32l23x/. -I/__w/zephyr/zephyr/subsys/rtio -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/zephyr/modules/cmsis/. -I/__w/zephyr/modules/hal/gigadevice/gd32l23x/cmsis/gd/gd32l23x/include -I/__w/zephyr/modules/hal/gigadevice/gd32l23x/standard_peripheral/include -I/__w/zephyr/modules/hal/gigadevice/include -I/__w/zephyr/modules/hal/gigadevice/common_include -Wshadow -fno-strict-aliasing -Werror -Os -imacros /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m23 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/opt/toolchains/zephyr-sdk-0.16.3/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/samples/sensor/sensor_shell=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_shell_stream.c.obj -MF zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_shell_stream.c.obj.d -o zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_shell_stream.c.obj -c /__w/zephyr/zephyr/drivers/sensor/sensor_shell_stream.c
/tmp/ccT67pVa.s: Assembler messages:
/tmp/ccT67pVa.s:613: Error: only lo regs allowed with immediate -- `mov ip,#0'
[110/156] Building C object modules/hal_gigadevice/CMakeFiles/hal_gigadevice.dir/__w/zephyr/modules/hal/gigadevice/gd32l23x/cmsis/gd/gd32l23x/source/system_gd32l23x.c.obj
[111/156] Building C object zephyr/drivers/timer/CMakeFiles/drivers__timer.dir/cortex_m_systick.c.obj
[112/156] Linking C static library zephyr/drivers/serial/libdrivers__serial.a
[113/156] Linking C static library zephyr/subsys/rtio/libsubsys__rtio.a
[114/156] Building C object modules/hal_gigadevice/CMakeFiles/hal_gigadevice.dir/__w/zephyr/modules/hal/gigadevice/gd32l23x/standard_peripheral/source/gd32l23x_gpio.c.obj
[115/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/banner.c.obj
[116/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/device.c.obj
[117/156] Building C object modules/hal_gigadevice/CMakeFiles/hal_gigadevice.dir/__w/zephyr/modules/hal/gigadevice/gd32l23x/standard_peripheral/source/gd32l23x_rcu.c.obj
[118/156] Building C object modules/hal_gigadevice/CMakeFiles/hal_gigadevice.dir/__w/zephyr/modules/hal/gigadevice/gd32l23x/standard_peripheral/source/gd32l23x_usart.c.obj
[119/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/errno.c.obj
[120/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/main_weak.c.obj
[121/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/busy_wait.c.obj
[122/156] Building C object zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/sensor_shell.c.obj
[123/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/kheap.c.obj
[124/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/fatal.c.obj
[125/156] Building C object zephyr/kernel/CMakeFiles/kernel.dir/init.c.obj
[126/156] Building C object zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/default_rtio_sensor.c.obj
FAILED: zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/default_rtio_sensor.c.obj 
ccache /opt/toolchains/zephyr-sdk-0.16.3/arm-zephyr-eabi/bin/arm-zephyr-eabi-gcc -DGD32L233 -DIRC32K_VALUE=32000 -DKERNEL -DPICOLIBC_DOUBLE_PRINTF_SCANF -D_FORTIFY_SOURCE=1 -D_POSIX_C_SOURCE=200809 -D__LINUX_ERRNO_EXTENSIONS__ -D__PROGRAM_START -D__ZEPHYR_SUPERVISOR__ -D__ZEPHYR__=1 -I/__w/zephyr/zephyr/include -I/__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated -I/__w/zephyr/zephyr/soc/arm/gigadevice/gd32l23x -I/__w/zephyr/zephyr/lib/posix/getopt/. -I/__w/zephyr/zephyr/soc/arm/gigadevice/common/. -I/__w/zephyr/zephyr/soc/arm/gigadevice/gd32l23x/. -I/__w/zephyr/zephyr/subsys/rtio -I/__w/zephyr/modules/hal/cmsis/CMSIS/Core/Include -I/__w/zephyr/zephyr/modules/cmsis/. -I/__w/zephyr/modules/hal/gigadevice/gd32l23x/cmsis/gd/gd32l23x/include -I/__w/zephyr/modules/hal/gigadevice/gd32l23x/standard_peripheral/include -I/__w/zephyr/modules/hal/gigadevice/include -I/__w/zephyr/modules/hal/gigadevice/common_include -Wshadow -fno-strict-aliasing -Werror -Os -imacros /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/zephyr/include/generated/autoconf.h -fno-common -g -gdwarf-4 -fdiagnostics-color=always -mcpu=cortex-m23 -mthumb -mabi=aapcs -mfp16-format=ieee --sysroot=/opt/toolchains/zephyr-sdk-0.16.3/arm-zephyr-eabi/arm-zephyr-eabi -imacros /__w/zephyr/zephyr/include/zephyr/toolchain/zephyr_stdint.h -Wall -Wformat -Wformat-security -Wno-format-zero-length -Wno-pointer-sign -Wpointer-arith -Wexpansion-to-defined -Wno-unused-but-set-variable -Werror=implicit-int -fno-pic -fno-pie -fno-asynchronous-unwind-tables -ftls-model=local-exec -fno-reorder-functions --param=min-pagesize=0 -fno-defer-pop -fmacro-prefix-map=/__w/zephyr/zephyr/samples/sensor/sensor_shell=CMAKE_SOURCE_DIR -fmacro-prefix-map=/__w/zephyr/zephyr=ZEPHYR_BASE -fmacro-prefix-map=/__w/zephyr=WEST_TOPDIR -ffunction-sections -fdata-sections --specs=picolibc.specs -D_POSIX_THREADS -std=c99 -MD -MT zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/default_rtio_sensor.c.obj -MF zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/default_rtio_sensor.c.obj.d -o zephyr/drivers/sensor/CMakeFiles/drivers__sensor.dir/default_rtio_sensor.c.obj -c /__w/zephyr/zephyr/drivers/sensor/default_rtio_sensor.c
/tmp/ccnw8Ls2.s: Assembler messages:
/tmp/ccnw8Ls2.s:[275](https://github.com/zephyrproject-rtos/zephyr/actions/runs/6383134587/job/17323294466?pr=60063#step:13:276)8: Error: only lo regs allowed with immediate -- `mov ip,#0'
ninja: build stopped: subcommand failed.

INFO    - /__w/zephyr/zephyr/twister-out/gd32l233r_eval/samples/sensor/sensor_shell/sample.sensor.shell/build.log
```

--- a/include/zephyr/arch/arm/asm_inline_gcc.h
+++ b/include/zephyr/arch/arm/asm_inline_gcc.h
@@ -61,7 +61,15 @@ static ALWAYS_INLINE unsigned int arch_irq_lock(void)
 		"mrs %0, BASEPRI;"
 		"msr BASEPRI_MAX, %1;"
 		"isb;"
-		: "=r"(key), "=r"(tmp)
+		: "=r"(key),
+#if defined(CONFIG_ARMV8_M_BASELINE)
+		/* armv8-m.baseline's mov is limited to registers r0-r7.
+		 * Let the compiler know we have this constraint on tmp.
+		 */
+		"=l"(tmp)
+#else
+		"=r"(tmp)
+#endif
 		: "i"(_EXC_IRQ_DEFAULT_PRIO)
 		: "memory");
 #elif defined(CONFIG_ARMV7_R) || defined(CONFIG_AARCH32_ARMV8_R) \

